server/controllers/finance/cost_center_allocation_registry.js

```diff
@@ -0,0 +1,116 @@
+/**
+* Cost Center Allocation Keys
+*
+* This controller exposes an API to the client for reading allocation keys
+*/
+// const _ = require('lodash');
+const db = require('../../lib/db');
+const Stepdown = require('../../lib/stepdown');
+const fiscal = require('./fiscal');
+const ccAllocationKeys = require('./cost_center_allocation_bases');
+
+async function fetch(session, params) {
+  const defaultCurrentYear = new Date();
+  const enterpriseCurrencyId = session.enterprise.currency_id;
+
+  const periods = {
```

Comment:
I don't think we should make these default.  If a programmer makes a typo, they will be very confused why their dates are not working.

Can we just require the user to submit a period?  And throw an error is no period is specified?